File: backend/src/main/java/org/jdk/project/service/config/ConfigCommandSupport.java
package org.jdk.project.service.config;

import org.jdk.project.exception.BusinessException;

final class ConfigCommandSupport {

  private ConfigCommandSupport() {}

  static Long requiredId(Long value, String message) {
    if (value == null) {
      throw new BusinessException(message);
    }
    return value;
  }

  static int defaultInt(Integer value, int fallback) {
    return value == null ? fallback : value;
  }

  static String defaultString(String value, String fallback) {
    return value == null ? fallback : value;
  }

  static void ensureUpdated(int updated, String message) {
    if (updated == 0) {
      throw new BusinessException(message);
    }
  }
}

File: backend/src/main/kotlin/org/jdk/project/service/config/ConfigCommandSupport.kt
package org.jdk.project.service.config

import org.jdk.project.exception.BusinessException

internal object ConfigCommandSupport {
    @JvmStatic
    fun requiredId(value: Long?, message: String): Long = value ?: throw BusinessException(message)

    @JvmStatic
    fun defaultInt(value: Int?, fallback: Int): Int = value ?: fallback

    @JvmStatic
    fun defaultString(value: String?, fallback: String): String = value ?: fallback

    @JvmStatic
    fun ensureUpdated(updated: Int, message: String) {
        if (updated == 0) {
            throw BusinessException(message)
        }
    }
}
